glTFLoader/Interface.cs

```diff
@@ -428,5 +429,289 @@ public static void SaveBinaryModel(this Gltf model, byte[] buffer, BinaryWriter
+        {
+            if (model == null) throw new ArgumentNullException(nameof(model));
+
+            byte[] binBufferData = null;
+
+            using (var memoryStream = new MemoryStream())
+            {
+                var bufferViews = new List<BufferView>();
+                var bufferData = new Dictionary<int, byte[]>();
+
+                if (model.BufferViews != null)
+                {
+                    foreach (var bufferView in model.BufferViews)
+                    {
+                        memoryStream.Align(4);
+
+                        var byteOffset = memoryStream.Position;
+
+                        byte[] data;
+                        if (!bufferData.TryGetValue(bufferView.Buffer, out data))
+                        {
+                            data = model.LoadBinaryBuffer(bufferView.Buffer, gltfFilePath);
+                            bufferData.Add(bufferView.Buffer, data);
+                        }
+
+                        memoryStream.Write(data, bufferView.ByteOffset, bufferView.ByteLength);
+
+                        bufferView.Buffer = 0;
+                        bufferView.ByteOffset = (int)byteOffset;
+                        bufferViews.Add(bufferView);
+                    }
+                }
+
+                if (model.Images != null)
+                {
+                    for (var index = 0; index < model.Images.Length; index++)
+                    {
+                        var byteOffset = memoryStream.Position;
+
+                        var data = model.OpenImageFile(index, gltfFilePath);
+                        data.CopyTo(memoryStream);
+
+                        var image = model.Images[index];
+                        image.BufferView = bufferViews.Count;
+                        image.MimeType = GetMimeType(image.Uri);
+                        image.Uri = null;
+
+                        bufferViews.Add(new BufferView
+                        {
+                            Buffer = 0,
+                            ByteOffset = (int)byteOffset,
+                            ByteLength = (int)data.Length,
+                        });
+                    }
+                }
+
+                if (bufferViews.Any())
+                {
+                    model.BufferViews = bufferViews.ToArray();
+
+                    model.Buffers = new[]
+                    {
+                        new glTFLoader.Schema.Buffer
+                        {
+                            ByteLength = (int)memoryStream.Length
+                        }
+                    };
+
+                    binBufferData = memoryStream.ToArray();
+                }
+            }
+
+            Interface.SaveBinaryModel(model, binBufferData, binaryWriter);
+
+        }
+
+        /// <summary>
+        /// Converts self contained GLB to glTF file and associated textures and data
+        /// </summary>
+        /// <param name="inputFilePath">Source</param>

```

Comment:
missing description for `inputFilePath` and `outputDirectoryPath`?

Reply:
```suggestion
        /// <param name="inputFilePath">glTF binary file (.glb) to unpack</param>
        /// <param name="outputDirectoryPath">Directory where the files will be extracted</param>
```